lifecycle/lifecycle.go

```diff
@@ -48,11 +49,29 @@ const (
 	Governor
 )
 
+//lint:ignore U1000 #1768 support for external clients
+func (t Type) String() string {
+	s := map[Type]string{
+		Unknown:     "Unknown",
+		Startup:     "Startup",
+		Shutdown:    "Shutdown",
+		Provisioned: "Provisioned",
+		Alive:       "Alive",
+		Governor:    "Governor",
+	}[t]
```

Comment:
This was better before when you had the maps external to the function, this way we allocate a bunch of strings every time we call this then dont use them.

A better way - if you want to do in the function like this would be to use a case statement on t and just return the string, then at least it doesnt allocate all the names for all the types  and the map each time

Reply:
I can't wait for golang to eventually realize people want Enums and do this natively.